### ISSUE TITLE
[operator][molecule] move to kubernetes.core ansible collection

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#4119]
There is a pretty big change hitting an ansible collection we heavily rely upon: https://github.com/ansible-collections/kubernetes.core/issues/32

I just noticed this because one of our CI machines installed the latest community.kubernetes and it bubbled up some errors in our molecule tests which I think are due to changes in that collection.

We should upgrade to kubernetes.core v2.0 both in molecule and the operator since community.kubernetes is deprecated. The operator work will be done separately - see https://github.com/kiali/kiali/issues/4121

This is the warning I see when I run with this now:

```
The community.kubernetes collection is being renamed to 
kubernetes.core. Please update your FQCNs to kubernetes.core instead. This 
feature will be removed from community.kubernetes in version 3.0.0.
```

**Comment by jmazzitelli** [comment]
I believe this is the cause of the error detected in our molecule tests when running with the latest kubernetes.core 2.0: https://github.com/ansible-collections/kubernetes.core/issues/9

**Comment by jmazzitelli** [comment]
We eventually will have to convert the operator ansible playbooks themselves, but we can't until the ansible operator base images pick up this new kubernetes.core collection.

In the meantime, we should move up our molecule ansible playbooks to use this new collection. It will move our molecule ansible playbooks forward to the latest code and will also provide us a headstart in finding what "gotchas" there are going to be when moving to this new collection. I already found one gotcha that I reported here: https://github.com/ansible-collections/kubernetes.core/issues/147
